lib/factory_bot/attribute_assigner.rb

```diff
@@ -53,7 +53,16 @@ def build_hash
     end
 
     def get(attribute_name)
-      @evaluator.send(attribute_name)
+      case obj = @evaluator.send(attribute_name)
```

Comment:
Thoughts on extracting instance access from mapping over the array? e.g.

```ruby
def extract_instance(obj)
  if obj.is_a?(FactoryBot::Evaluator)
    obj.instance
  else
    obj
  end
end

def get(attribute_name)
  obj = @evaluator.send(attribute_name)

  if obj.is_a?(Array)
    obj.map { |o| extract_instance(o) }
  else
    extract_instance(obj)
  end
end
```

Reply:
I forget if there was a reason I didn't extract already. Maybe I was waiting for a third repitition?

Reply:
I will say, in the ideal world, we would delete the hard check in rails on object type and do duck typing instead, which would negate the need for this if statement.

Reply:
After thinking for a few days, I want to leave it as is for now as I think the extraction makes the code slightly more confusing and it is unknown if this will be useful in the future.